.github/workflows/migrate.yml

```
@@ -0,0 +1,66 @@
+        run: |
+          npm install -g pnpm@9.15.3 && pnpm install
+
+      - name: Create .env file
+        run: |
+          if [ "${{ github.event.inputs.environment }}" == "production" ]; then
+            echo "DATABASE_URL=${{ secrets.PRODUCTION }}" > .env
+          else
+            echo "DATABASE_URL=${{ secrets.STAGING }}" > .env
+
+      - name: Backup Database before migration
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC1046:error:2:1: Couldn't find 'fi' for this 'if'

(shellcheck)